online-bookstore/src/main/java/com/example/onlinebookstore/services/book/BookServiceImpl.java

```diff
@@ -125,4 +132,20 @@ private void updateBookFields(BookDTO bookDTO, BookEntity editedBook) {
         editedBook.setQrCode(bookDTO.getQrCode());
         editedBook.setInventory(bookDTO.getInventory());
     }
+
+    private GenreEntity checkIfGenreExistsInDBAndMerge(GenreEntity genreToBeSaved) {
+        List<GenreEntity> iterateRepositoryList = genreRepository.findAll();
+        boolean genreAlreadyExists = false;
+        for (GenreEntity genre : iterateRepositoryList) {
+            if (genre.getName().equals(genreToBeSaved.getName())) {
+                genreAlreadyExists = true;
+                genreToBeSaved = genre;
+                break;
+            }
+        }
+        if (!genreAlreadyExists) {
+            genreRepository.save(genreToBeSaved);
```

Comment:
Can we use this action inside the for loop, maybe? When we discover that there is a genre that doesn't exists in the database, we save it without other checks an boolean statements.

Reply:
Yes again, my brain on low power had to think of something that was easy for my brain, there is no reason to have two if statements, decision making can be done with only one if.